scvi/models/log_likelihood.py

```diff
@@ -256,21 +256,17 @@ def log_nb_positive(x, mu, theta, eps=1e-8):
     theta: inverse dispersion parameter (has to be positive support) (shape: minibatch x genes)
     eps: numerical stability constant
     """
-    if theta.ndimension() == 1:
-        theta = theta.view(
-            1, theta.size(0)
+    if log_theta.ndimension() == 1:
+        log_theta = log_theta.view(
+            1, log_theta.size(0)
         )  # In this case, we reshape theta for broadcasting
 
-    log_theta_mu_eps = torch.log(theta + mu + eps)
-
-    res = (
-        theta * (torch.log(theta + eps) - log_theta_mu_eps)
-        + x * (torch.log(mu + eps) - log_theta_mu_eps)
-        + torch.lgamma(x + theta)
-        - torch.lgamma(theta)
-        - torch.lgamma(x + 1)
+    theta = torch.exp(log_theta)
+    res = (theta * log_theta + x * log_mu
+           + torch.lgamma(x + theta)
+           + torch.lgamma(x + theta)
+           - torch.lgamma(theta)
```

Comment:
is this correct? There should probably be a log sum exp for the `log_theta_mu_eps` term in the old code.

Reply:
The beauty here is that `log_theta_mu_eps` is no longer needed, since theta is already in log units, no need for the back and forth transform.

Reply:
This is the function we optimize, `r=theta` here and `m=mu`. Therefore you're going to have a `log(m + r)` term. If `mu` and `r` are on the log scale you'll have to do a logsumexp computation.
![Screen Shot 2020-04-28 at 10 50 47 AM](https://user-images.githubusercontent.com/10859440/80520333-203bf780-893e-11ea-851d-34631d17f6aa.png)

Reply:
Awesome. Thanks for the reference.